rolediscordbot.py

```diff
@@ -0,0 +1,110 @@
+import discord
+from discord.ext import commands
+
+intents = discord.Intents.default()
+intents.reactions = True
+intents.message_content = True
+intents.members = True
+
+client = commands.Bot(command_prefix="!", intents=intents)
+role_selection_message = None
+
+@client.event
+async def on_ready():
+    global role_selection_message
+    print(f'Logged in as {client.user.name}')
+    
+    # Find "roles-selection" channel
+    role_selection_channel = discord.utils.get(client.guilds[1].text_channels, name="roles-selection")
+
+    # Checking for channel
+    if role_selection_channel:
+        # Check if there's an existing role selection message
+        async for message in role_selection_channel.history(limit=1):
+            if message.author == client.user:
+                role_selection_message = message
+                break
+        
+        if role_selection_message:
+            print(f'Found existing role selection message: {role_selection_message.jump_url}')
```

Comment:
consider changing print statements to log statements: https://docs.python.org/3/howto/logging.html